### PR TITLE
PAR - Hanane done

### PR DESCRIPTION
**Pull request by HananeDUP**

### CODE REVIEW

File: starter-code/src/App.js [hunk 1]
```diff
@@ -1,21 +1,234 @@
...
+import foods from './foods.json'
+
+class FoodTag extends Component {
+  constructor(props) {
+    super(props)
+    this.state = { qty: 1 }
+  }
+
+  handleChange(event) {
+    this.setState({ qty: parseInt(event.target.value) })
```
Comment (lucamerzi) [inline starter-code/src/App.js]: It is a good practice to pass radix with parseInt like so:
```js
parseInt(string, radix)
```
Also, you have an alternative to parseInt(), which is Number()

File: starter-code/src/App.js [hunk 2]
```diff
@@ -1,21 +1,234 @@
...
+    this.setState({ listFoods: arrayFood })
+  }
+
+  searchFilter(food) {
+    return food.name.includes(this.state.searchText)
+  }
+
+  addAppQuantity(name, val) {
+    // var arraySelected = {};
+    var selectedFoodCopy = this.state.selectedFood
```
Comment (lucamerzi) [inline starter-code/src/App.js]: Technically, you are not making a copy here. In order to make a copy of the state you can use the slice method like so:
```js
var selectedFoodCopy = this.state.selectedFood.slice()
```
But bravo for using the best practices of React (always making copy of state and never modifying it directly) 👏

**Comment by ta-webpt-par** [comment]
Congrats on finishing the exo. 👍  This is one of the most challenging of the module! It sums up a lot of React features we've seen so far. Some points that came up during the exo:
```js

// You have an alternative to using parseInt() (which, in React.js is going to trigger some warnings):
Number()


// Props are immutable: never ever change this.props
// The following examples are INCORRECT:
this.props.text = "something";      // WRONG!
this.props = {};                    // WRONG!
this.props.myNewProp = "something"  // WRONG!


// State can change, but it SHOULD NEVER be modified directly. The correct way to modify state is using 
this.setState({ });
// Technically, as a best practice, try to never modify the state object directly, by making copies:
var selectedFoodCopy = this.state.selectedFood.slice()


// When a setState depends on previous state, you can use a function parameter:
this.state = { counter: 1 }
this.setState((prevState, props) => {
    return {
        counter: prevState.counter + 1
    }
})


// setState IS ASYNCHRONOUS
this.setState({firstName: "Luca"});
console.log(this.state.firstName): // Won't be updated yet

// Use a callback function as a workaround:
this.setState({firstName: "Luca"}, () => {
    console.log(this.state)
})

```

**Comment by HananeDUP** [comment]
Thanks Lucas for your comments! Interesting

Le jeu. 25 oct. 2018 à 00:03, ta-webpt-par <notifications@github.com> a
écrit :

**Pull request closed without merging**